### PR TITLE
Remove unused cni code

### PR DESCRIPTION
**Pull request by amshinde**
We dont really call CNI from Kata, this is done by the CRI layer.
All of the CNI code is unused.

### CODE REVIEW

**Comment by amshinde** [comment]
cc @sboeuf @egernst

File: virtcontainers/network.go [hunk 1]
```diff
@@ -582,8 +582,6 @@ func newNetwork(networkType NetworkModel) network {
 	switch networkType {
 	case NoopNetworkModel:
 		return &noopNetwork{}
-	case CNINetworkModel:
```
Comment (sboeuf) [inline virtcontainers/network.go]: You forgot to remove the CNINetworkModel type from the list.

**Comment by amshinde** [comment]
> A follow up PR might be to change the network Go interface into a simple class. I don't think there is any reason to have several implementations of network

Yes, I realised that we need to simplify this, I was planning to open a separate issue for this to address this . But looks like you have already opened one :)

> We need to move the OCI hook to the CLI level as the Kata API should not be aware about something OCI specific.

Yes, I remember we have talked about this in the past. I agree the OCI hook implementation needs to be  refactored and the hooks need to be called from the CLI instead. This will also help address this issue: https://github.com/kata-containers/runtime/issues/264

**Comment by egernst** [comment]
Thanks - this, at the very least, will limit confusion going forward.  thanks archana.  lgtm minus the need to still remove the CNINetworkModel case statement @sboeuf pointed out.

**Comment by grahamwhaley** [comment]
CIs not happy. From the 16.04:
```
=== RUN   TestStartStopSandboxHyperstartAgentSuccessfulWithCNMNetwork
--- FAIL: TestStartStopSandboxHyperstartAgentSuccessfulWithCNMNetwork (0.01s)
	cc_proxy_mock.go:80: [CCProxyMock] listening on /tmp/cc-proxy-test920369036/cc-proxy-test.sock
	cc_proxy_mock.go:80: [CCProxyMock] Client connected
	cc_proxy_mock.go:80: [CCProxyMock] Register VM
	cc_proxy_mock.go:80: [CCProxyMock] Error serving client : EOF
	cc_proxy_mock.go:80: [CCProxyMock] Client closed connection
	cc_proxy_mock.go:80: [CCProxyMock] Client connected
	cc_proxy_mock.go:80: [CCProxyMock] Attach VM
	cc_proxy_mock.go:80: [CCProxyMock] Hyper command
	cc_proxy_mock.go:80: [CCProxyMock] Error serving client : EOF
	cc_proxy_mock.go:80: [CCProxyMock] Client closed connection
	cc_proxy_mock.go:80: [CCProxyMock] Client connected
	cc_proxy_mock.go:80: [CCProxyMock] Attach VM
	api_test.go:1380: Can not move from running to running
	cc_proxy_mock.go:80: [CCProxyMock] Closing client connection
	cc_proxy_mock.go:80: [CCProxyMock] Error serving client : EOF
	cc_proxy_mock.go:80: [CCProxyMock] Client closed connection
	cc_proxy_mock.go:80: [CCProxyMock] Stopped
```

**Comment by jodh-intel** [comment]
*Less is more* - love it!

lgtm once the extra tidyup mentioned above is done.

**Comment by amshinde** [comment]
@grahamwhaley Yes, I am working on figuring out why the test is failing. If I run the test alone with:
go test -run TestStartStopSandboxHyperstartAgentSuccessfulWithCNMNetwork
it passes.

**Comment by bergwolf** [comment]
LGTM! Glad we finally drop the confusing bits ;)

**Comment by jodh-intel** [comment]
Hi @amshinde - please rebase to resolve the conflict.

**Comment by sboeuf** [comment]
@chavafg have we merged something wrong? I mean, is the CI not stable for some reason?
I am asking because I don't understand why all the tests are failing here.

**Comment by amshinde** [comment]
@sboeuf This needs to merged to fix CI for now :
https://github.com/kata-containers/runtime/pull/583

Without that commit we are always building qemu from source, and the soak tests are failing when we build from source. That commit will cause CI to use the packaged version of qemu. See this for more background:
https://github.com/kata-containers/tests/pull/610#issuecomment-413055159

**Comment by sboeuf** [comment]
@amshinde #583 merged ! Please repush !

**Comment by amshinde** [comment]
@sboeuf Thanks! Repushed.

**Pull request merged**